aws/resource_aws_glue_trigger.go

```diff
@@ -0,0 +1,457 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlueTrigger() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueTriggerCreate,
+		Read:   resourceAwsGlueTriggerRead,
+		Update: resourceAwsGlueTriggerUpdate,
+		Delete: resourceAwsGlueTriggerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"actions": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arguments": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+						"job_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"timeout": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
```

Comment:
> it feels like this should be set by default in TF Core when a fields required?

Some resource attributes (not here) use `""` to denote removing or zero-ing out something currently. When Terraform 0.12 is released, maybe we can become more strict, but for now the validation is required.

>  we generally sort Required -> Optional -> Computed in the schema

Yikes, there are probably hundreds of AWS resources that need to be updated then. It would be nice if there was documentation or linting for this.